public/src/test/java/com/rackspace/salus/telemetry/api/web/client/TestMonitorControllerTest.java

```diff
@@ -0,0 +1,136 @@
+import com.rackspace.salus.telemetry.api.services.TestMonitorAndEventTaskService;
+import com.rackspace.salus.telemetry.api.web.TestMonitorController;
+import com.rackspace.salus.telemetry.api.web.TraceResponseFilter;
+import java.util.List;
+import java.util.Map;
+import org.apache.commons.lang3.RandomStringUtils;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.MediaType;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+import uk.co.jemos.podam.api.PodamFactory;
+import uk.co.jemos.podam.api.PodamFactoryImpl;
+
+@RunWith(SpringRunner.class)
+@WebMvcTest(controllers = TestMonitorController.class)
+public class TestMonitorControllerTest {
+
+  @Autowired
+  MockMvc mockMvc;
+
+  @Autowired
+  ObjectMapper objectMapper;
+
+  private PodamFactory podamFactory = new PodamFactoryImpl();
+
+  @Autowired
+  TestMonitorController testMonitorController;
+
+  @MockBean
+  TraceResponseFilter traceResponseFilter;
+
+  @MockBean
+  ApiPublicProperties apiPublicProperties;
+
+  @MockBean
+  TestMonitorAndEventTaskService testMonitorAndEventTaskService;
+
+  @MockBean
+  ServicesProperties servicesProperties;
+
+  @Test
+  public void testCreateTestMonitorAndEventTask_Success() throws Exception {
+    String tenantId = RandomStringUtils.randomAlphabetic(8);
+    TestMonitorAndEventTaskRequest testMonitorAndEventTaskRequest = objectMapper
+        .readValue(readContent("testCreateTestMonitorAndEventTask_req.json"),
+            TestMonitorAndEventTaskRequest.class);
+
+    TestMonitorAndEventTaskResponse testMonitorAndEventTaskResponse = podamFactory
+        .manufacturePojo(TestMonitorAndEventTaskResponse.class);
+
+    final TestTaskResult testTaskResultExpected = new TestTaskResult()
+        .setEvents(List.of(
+            new EventResult()
+                .setData(
+                    new EventData()
+                        .setSeries(List.of(
+                            new SeriesItem()
+                                .setName(testMonitorAndEventTaskRequest.getTask().getMeasurement())
+                        ))
+                )
+                .setLevel("CRITICAL")
+        ))
+        .setStats(
+            new Stats()
+                .setNodeStats(Map.of("alert2", Map.of("crits_triggered", 1)))
+        );
+    testMonitorAndEventTaskResponse.setTask(testTaskResultExpected);
+
+    when(testMonitorAndEventTaskService.getTestMonitorAndEventTask(anyString(), any()))
+        .thenReturn(testMonitorAndEventTaskResponse);
+
+    mockMvc.perform(
+        MockMvcRequestBuilders.post("/v1.0/api/tenant/{tenantId}/test-monitor-event-task", tenantId)
```

Comment:
MockMvc doesn't make use of the servlet context shift, so needs to be
```suggestion
        MockMvcRequestBuilders.post("/tenant/{tenantId}/test-monitor-event-task", tenantId)
```